### PR TITLE
clean up of methods in MainMenu

### PR DESCRIPTION
**Pull request by euloong**
- Used .trim() to clean user input of trailing whitespaces in manageOptions, an unintentional effect of .equals() appears to clean up the input but it's best to intentionally manage the behaviour of the code.
- Replaced counter = 0 with a boolean condition to manage the while loop as I only need it to be on or off.
- Changed arrayLists to private from no descriptor.
- Moved "Please select an option from the following:" out of the options ArrayList.
- Updated for loop notation in showOptions to for-each loop.
- Changed if ((this.books.size() - count) == 0) to if (this.books.size() == count)

### CODE REVIEW

File: src/com/twu/biblioteca/MainMenu.java [hunk 1]
```diff
@@ -11,14 +11,15 @@
     private final String goodBye = "Goodbye!";
     private final String checkOutBookReference = "Enter the number of the book you want to checkout:";
     private final String noBooks = "Sorry, no books left!";
+    private final String selectOption = "Please select an option from the following:";
 
-    ArrayList<String> options = new ArrayList<>(Arrays.asList(
-            "Please select an option from the following:",
+    private ArrayList<String> options = new ArrayList<>(Arrays.asList(
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:16]: :+1:

File: src/com/twu/biblioteca/MainMenu.java [hunk 2]
```diff
@@ -11,14 +11,15 @@
...
+            //"Please select an option from the following:",
             "1. List of books",
             "2. Checkout a book",
             "3. Quit"));
 
-   ArrayList<Book> books = new ArrayList<>(Arrays.asList(
+   private ArrayList<Book> books = new ArrayList<>(Arrays.asList(
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:22]: :+1:

File: src/com/twu/biblioteca/MainMenu.java [hunk 3]
```diff
@@ -45,26 +46,26 @@ public void showNoBooksMessage() {
 
 
     public void showOptions() {
-        for (int i = 0; i < options.size(); i++) {
-            System.out.println(options.get(i));
+        System.out.print(selectOption + "\n");
+        for (String option : options) {
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:50]: Even though we've not saved any lines of code here, not having to manage (i.e., initialise, increment, check) the `i` index makes this change worthwhile.

File: src/com/twu/biblioteca/MainMenu.java [hunk 4]
```diff
@@ -45,26 +46,26 @@ public void showNoBooksMessage() {
...
         }
         System.out.print(">");
     }
 
     public void manageOptions() {
-        int counter = 0;
         Scanner optionsScanner = new Scanner(System.in);
-
-        while (counter == 0) {
+            while (true) {
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:58]: Yes... and no.  Using infinite loops and having to explicitly break out of them (in this case, using `System.exit`) is generally considered a bit lazy.  I like using a boolean to control the loop, but could it be a variable which is changed when the user selects `3` instead?

File: src/com/twu/biblioteca/MainMenu.java [hunk 5]
```diff
@@ -45,26 +46,26 @@ public void showNoBooksMessage() {
...
                 showOptions();
-            } else if (userInput.equals("2")) {
+            } else if (cleanUserInput.equals("2")) {
                 showBookList();
                 manageBookList();
-            } else if (userInput.equals("3")) {
+            } else if (cleanUserInput.equals("3")) {
                 showGoodbyeMessage();
                 System.exit(0);
             } else {
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:71]: A question for Future You: Is there a point at which these ifs would be better implemented as a `switch`?

File: src/com/twu/biblioteca/MainMenu.java [hunk 6]
```diff
@@ -105,7 +106,7 @@ public int countCheckedOutBooks() {
 
     public void manageBookList() {
         int count = countCheckedOutBooks();
-      if ((this.books.size() - count) == 0) {
+        if (this.books.size() == count) {
```
Comment (andeemarks) [inline src/com/twu/biblioteca/MainMenu.java:109]: :+1: 

You could _also_ consider getting rid of the `count` variable and calling `countCheckedOutBooks` instead, as the method name is clearer than the variable itself.

Comment (euloong) [reply src/com/twu/biblioteca/MainMenu.java:109]: Ok, thanks.

**Pull request merged**